src/Hooks.ts

```diff
@@ -17,7 +17,10 @@ import {
   EventType,
 } from 'react-navigation';
 
-export function useNavigation<S>(): NavigationScreenProp<S & NavigationRoute> {
+export function useNavigation<S, P = NavigationParams>(): NavigationScreenProp<
```

Comment:
looks good for the specific usecase of params, but for #42 we should also probably remove the hardcoded "NavigationScreenProp" reference because it won't expose the custom navigator actions (like popToTop etc)

I prefer to have one PR fix both the params usecase and custom navigator actions usecases, because if we release a fix for #42 later, we're probably going to break the TS types retrocompatibility so it's likely to be annoying for the users.